# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/nsICookieManager.idl
# Commit: 201f0d5fabd6
# Full Hash: 201f0d5fabd61c8eebe7e27e8597021d3618ddea
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2021-03-29 15:25:22
# Regressor Bug: 1695050
# File Overlap Count: 1
# Description:
#   Bug 1695050 - Part 2: Modify CookieJarSettings::Create() to be able to creat cookieJarSettings according to the browsing mode. r=dimi,preferences-reviewers
#   
#   This patch modifies the current CookieJarSettings::Create() function.
#   It removes the current function and adds two variants. One takes the
#   nsIPrincipal as input and another takes the enum value. The new
# ==============================================================================

diff -r 59cda3c92f48 -r 201f0d5fabd6 netwerk/cookie/nsICookieManager.idl
--- a/netwerk/cookie/nsICookieManager.idl	Mon Mar 29 11:01:50 2021 +0000
+++ b/netwerk/cookie/nsICookieManager.idl	Mon Mar 29 11:01:51 2021 +0000
@@ -45,11 +45,14 @@
   readonly attribute Array<nsICookie> sessionCookies;
 
   /**
-   * Returns current effective value of the "network.cookie.cookieBehavior".
+   * Returns current effective value of the cookieBehavior. It will return the
+   * different pref according to the aIsPrivate. If aIsPrivate is true, it will
+   * return the pref "network.cookie.cookieBehavior". Otherwise, it will return
+   * the pref "network.cookie.cookieBehavior.pbmode"
    */
-  readonly attribute uint32_t cookieBehavior;
+  uint32_t getCookieBehavior(in boolean aIsPrivate);
   %{C++
-    static uint32_t GetCookieBehavior();
+    static uint32_t GetCookieBehavior(bool aIsPrivate);
   %}
 
   /**